fili-core/src/main/java/com/yahoo/bard/webservice/web/DataApiRequest.java

```diff
@@ -290,6 +327,24 @@ public DataApiRequest(
     }
 
     /**
+     * Times the execution of the specified producer, making sure to stop timing in a finally clause.
+     *
+     * @param generator  The producer to be timed
+     * @param timerName  The name to use for the timer
+     * @param <T>  The type of the object produced
+     *
+     * @return The value returned by the generator
+     */
+    private <T> T timeGenerator(Producer<T> generator, String timerName) {
```

Comment:
I agree 100% that making the RequestLog annotation-based would be a Good Thing(TM).

I haven't actually pushed the timings into the methods. I'll do that now.